R/tfb-wavelet-utils.R

```diff
@@ -29,28 +29,52 @@ interpolate_arg <- function(arg_list) {
 
 fit_wavelet <- function(data, Z, penalized, glmnet_args) {
   eval_list <- split(data$value, data$id)
-  arg_list <- split(data$arg, data$id)
+  arg_list <- unique(data$arg)
+  
+  Z <- cbind(Z, arg_list)
+  Z <- scale(Z, center = FALSE)
   
-  correction <- map2(arg_list, eval_list, function(x, y) remove_slope(x,y))
+  if (!penalized) {
+    coefs <- map(eval_list, 
+                  function(x) {
+                    # Least squares fit, directly computed
+                    unname(lsfit(Z, x, intercept = TRUE)$coef)})
+  } else {
+    coefs <- map(eval_list, 
+                  function(x, y) {
+                    temp_model <- do.call(glmnet::cv.glmnet, c(list(Z, y), 
+                                                               glmnet_args))
+                    as.numeric(coefficients(temp_model))
+                  }
+    )
+  }
+  coefs
+}
+
+
+fit_wavelet_irr <- function(data, Z, penalized, glmnet_args) {
```

Comment:
This is not really in use, but it makes fit_wavelet a little bit more flexible and faster, because the matrix doesn't need to be combined every iteration.